tests/components/stream/test_hls.py

```diff
@@ -298,17 +298,18 @@ async def test_hls_max_segments(hass, hls_stream, stream_worker_sync):
     )
 
     # Fetch the actual segments with a fake byte payload
-    with patch(
-        "homeassistant.components.stream.hls.get_m4s", return_value=b"fake-payload"
-    ):
-        # The segment that fell off the buffer is not accessible
-        segment_response = await hls_client.get("/segment/1.m4s")
-        assert segment_response.status == 404
+    for segment in hls.get_segments():
+        segment.init = INIT_BYTES
```

Comment:
Did this already get initialized on line 284?

Reply:
Good catch, I will remove it